static/js/message_events.js

```diff
@@ -484,14 +483,13 @@ export function update_messages(events) {
         // edit.  Doing so could save significant work, since most
         // topic edits will not match the current topic narrow in
         // large organizations.
-        if (!changed_narrow && message_lists.current === message_list.narrowed) {
-            message_list.narrowed.update_muting_and_rerender();
+        if (!changed_narrow && message_lists.current.is_current_view()) {
```

Comment:
This is I think migrated wrong; `message_lists.current.is_current_view()` should always be true.

Reply:
I think you want to check `message_lists.current.narrowed` in all of these places.

Reply:
(Or are you trying to exclude the case where recent topics is visible? The bug I see is that `message_lists.current.is_current_view()` is true when viewing "All messages").

Reply:
```js
    is_current_view() {
        // No need to update message_list if recent topics is visible.
        if (recent_topics_util.is_visible()) {
            return false;
        }
        // narrow_state.active is false for both recent_topics and home.
        if (narrow_state.active()) {
            return this.narrowed;
        }
        // Current narrow is home, return true if this list is home.
        return this.table_name === "zhome";
    }
```

If the message_list is the message_list.home, then it will return true, otherwise not when we are narrowed to all messages.

Reply:
If we previously were checking `message_list.current === message_list.narrowed`, that should be equivalent to checking if `message_lists.current.narrowed` is true, since we set that boolean correctly when creating a message list.

Reply:
Ok, updated!